src/dialects/snowflake/query-generator-typescript.ts

```diff
@@ -13,4 +17,32 @@ export class SnowflakeQueryGeneratorTypeScript extends AbstractQueryGenerator {
     // TODO [+snowflake-sdk]: check if this is the correct implementation
     return `SELECT '' FROM DUAL`;
   }
+
+  removeIndexQuery(tableName: TableNameOrModel, indexNameOrAttributes: string | string[], options: RemoveIndexQueryOptions) {
+    // TODO [+snowflake-sdk]: check if this is the correct implementation
+    if (options) {
+      rejectInvalidOptions(
+        'removeIndexQuery',
+        this.dialect.name,
+        REMOVE_INDEX_QUERY_SUPPORTABLE_OPTIONS,
+        new Set(),
+        options,
+      );
+    }
+
+    let indexName: string;
+    const table = this.extractTableDetails(tableName);
+    if (Array.isArray(indexNameOrAttributes)) {
+      indexName = generateIndexName(table, { fields: indexNameOrAttributes });
+    } else {
+      indexName = indexNameOrAttributes;
+    }
+
+    return joinSQLFragments([
+      'DROP INDEX',
+      this.quoteIdentifier(indexName),
+      'ON',
+      this.quoteTable(tableName),
+    ]);
```

Comment:
```suggestion
    return `DROP INDEX ${this.quoteIdentifier(indexName)} ON ${this.quoteTable(tableName)}`;
```
If there's no options or it's not too long I think we do not need to use `joinSQLFragments`. This is also in line with mysql